src/Security/Authentication/OpenIdConnect/src/OpenIdConnectHandler.cs

```diff
@@ -1320,5 +1341,146 @@ private OpenIdConnectProtocolException CreateOpenIdConnectProtocolException(Open
             ex.Data["error_uri"] = errorUri;
             return ex;
         }
+
+        protected override IEnumerable<Endpoint> GetEndpoints()
+        {
+            List<Endpoint> endpoints = new List<Endpoint>();
+
+            var getHandleEndpoint = CreateEndpoint<OpenIdConnectHandler>(Options.CallbackPath,
+               "AuthenticationHandlerGet" + Scheme.Name,
+               "GET",
+               async handler =>
+               {
+                   var result = await handler.HandleEndpointGetAsync();
+                   await handler.HandleAsync(result);
+               });
+
+            endpoints.Add(getHandleEndpoint);
+
+
+            var postHandleEndpoint = CreateEndpoint<OpenIdConnectHandler>(Options.CallbackPath,
+              "AuthenticationHandlerPost" + Scheme.Name,
+              "POST",
+              async handler =>
+              {
+                  var result = await handler.HandleEndpointPostAsync();
+                  await handler.HandleAsync(result);
+              });
+
+            endpoints.Add(postHandleEndpoint);
+
+            if (Options.RemoteSignOutPath.HasValue)
+            {
+                var getSignOutEndpoint = CreateEndpoint<OpenIdConnectHandler>(Options.RemoteSignOutPath,
+                    "RemoteSignOutPathGet" + Scheme.Name,
+                    "GET",
+                    x => x.HandleGetRemoteSignOutEndpoint());
+
+                endpoints.Add(getSignOutEndpoint);
+
+                var postSignOutEndpoint = CreateEndpoint<OpenIdConnectHandler>(Options.RemoteSignOutPath,
+                    "RemoteSignOutPathPost" + Scheme.Name,
+                    "GET",
+                    x => x.HandlePosttRemoteSignOutEndpoint());
+
+                endpoints.Add(getSignOutEndpoint);
+            }
+
+            if (Options.SignedOutCallbackPath.HasValue)
+            {
+                var getSignOutEndpoint = CreateEndpoint<OpenIdConnectHandler>(Options.SignedOutCallbackPath,
+                    "SignedOutCallbackPathGet" + Scheme.Name,
+                    "GET",
+                    x => x.HandleSignOutCallbackAsync());
+
+                endpoints.Add(getSignOutEndpoint);
+            }
+
+            return endpoints;
+        }
+
+        protected virtual async Task<HandleRequestResult> HandleEndpointGetAsync()
+        {
+            Logger.EnteringOpenIdAuthenticationHandlerHandleRemoteAuthenticateAsync(GetType().FullName);
+
+            OpenIdConnectMessage authorizationResponse;
+
+            if (string.Equals(Request.Method, "GET", StringComparison.OrdinalIgnoreCase))
```

Comment:
`HttpMethods.IsGet`? 🤔

Reply:
This was the current code and I didn't change it, but actually this check is no longer needed since this method is for `GET` endpoint.